### ISSUE TITLE
Puma-dev hard locking

### ISSUE THREAD

**Issue by vimalloc** [puma/puma-dev#207]
About once a day I'll run into a situation where my application stops responding to requests through puma-dev. When this happens, I am unable to stop/restart puma-dev. Doing a `puma-dev -stop` results in the message `Gracefully stopping, waiting for requests to finish` to be logged, then it hangs until I eventually `kill -9` the puma-dev process. 

I'm running `Version: v0.12 (go1.10.2)` on macos. Any ideas on what's going on here or how I could troubleshoot this further?

**Comment by nonrational** [comment]
I believe this is a duplicate of #59.

You can try building with https://github.com/puma/puma-dev/pull/175 which, in my experience, fixes the issue.

**Comment by vimalloc** [comment]
Update in case anyone else runs into this. Running with the patch applied did not fix the issue for me, but since I changed the number of threads to 1 it has been working flawlessly. I think we made some chanages in our app in regards to stream response vs chunking responses around the time I started seeing this issue, so I'm thinking it might be related to that? Either way, it's now stable for me so I haven't dug into it any further.
